curator-client/src/main/java/org/apache/curator/utils/InjectSessionExpiration.java

```diff
@@ -94,7 +89,7 @@ public static void injectSessionExpiration(ZooKeeper zooKeeper)
             Object eventThread = eventThreadField.get(clientCnxn);
             queueEventMethod.invoke(eventThread, event);
             queueEventOfDeathMethod.invoke(eventThread);
```

Comment:
Why it is important to queue event of death immediately? It will be done for us soon by this flow:
ConnectionState.process -> handleExpiredSession -> ... -> ClientCnxn.disconnect ->...
- because we just queued this WatchedEvent.
Same thing for wakeupCnxnMethod.invoke.

Reply:
TBH I just wanted to make the minimal change needed to get this to work. It's been this way a very long time.